powerscale/provider/smb_server_settings_resource.go

```diff
@@ -0,0 +1,509 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	powerscale "dell/powerscale-go-client"
+	"fmt"
+	"strings"
+
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/constants"
+	"terraform-provider-powerscale/powerscale/helper"
+	"terraform-provider-powerscale/powerscale/models"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &SmbServerSettingsResource{}
+	_ resource.ResourceWithConfigure   = &SmbServerSettingsResource{}
+	_ resource.ResourceWithImportState = &SmbServerSettingsResource{}
+)
+
+// NewSmbServerSettingsResource is a helper function to simplify the provider implementation.
+func NewSmbServerSettingsResource() resource.Resource {
+	return &SmbServerSettingsResource{}
+}
+
+// SmbServerSettingsResource is the resource implementation.
+type SmbServerSettingsResource struct {
+	client *client.Client
+}
+
+// Metadata defines the resource type name.
+func (r *SmbServerSettingsResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_smb_server_settings"
+}
+
+// Schema defines the schema for the resource.
+func (r *SmbServerSettingsResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `This resource is used to manage the SMB Server Settings of PowerScale Array. We can Create, Update and Delete the SMB Server Settings using this resource.  
+		Note that, SMB Server Settings is the native functionality of PowerScale. When creating the resource, we actually load SMB Server Settings from PowerScale to the resource.`,
+		Description: `This resource is used to manage the SMB Server Settings of PowerScale Array. We can Create, Update and Delete the SMB Server Settings using this resource.  
+		Note that, SMB Server Settings is the native functionality of PowerScale. When creating the resource, we actually load SMB Server Settings from PowerScale to the resource.`,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				Description:         "ID of SMB Server Settings.",
+				MarkdownDescription: "ID of SMB Server Settings.",
+			},
+			"scope": schema.StringAttribute{
+				Computed:            true,
```

Comment:
Can we remove computed? Will scope be returned from API

Reply:
Scope is not returned from the API.
It is a query parameter for the API that can have 3 possible values "effective", "user" and "default".(If not specified, it is considered as "effective")

So, have kept 'scope' as both "Computed" and "Optional" as true in the schema.
If specified in the configuration file, it is picked up as it is.
Else if not specified, it is computed as "effective".

![1](https://github.com/dell/terraform-provider-powerscale/assets/44159729/86f68735-04ef-41b3-bf29-dfe920b60c10)

![2](https://github.com/dell/terraform-provider-powerscale/assets/44159729/e35bc6e8-bc3f-4982-adc8-89ca4e617b1e)

Reply:
OK, got it. That means you set the scope in the state at last. Thanks for clarifying!